src/watch.rs

```diff
@@ -109,17 +115,35 @@ impl Watch {
     /// Extend the watch list with an additional list of paths.
     /// Note: Watch maintains a list of already watched paths, and
     /// will not add duplicates.
-    pub fn extend(&mut self, paths: &[PathBuf]) -> Result<(), notify::Error> {
+    pub fn extend(&mut self, paths: Vec<PathBuf>) -> Result<(), notify::Error> {
```

Comment:
```suggestion
    pub fn extend(&mut self, paths: IntoIterator<Item=PathBuf>) -> Result<(), notify::Error> {
```
I tend to abstract to a trait when I change types like this

Reply:
We’re not exposing it as library, so abstractions like that are not really needed (they only complicated the code somewhat).